submissions/factorio_draftsman/main.md

```diff
@@ -0,0 +1,200 @@
+    x = 0
+    y = 0
+    combinator = ConstantCombinator(direction=Direction.SOUTH)
+
+    # Iterate over every item in order:
+    for item in items.raw:
+        # Ignore hidden items/entities
+        if "flags" in items.raw[item]:
+            if "hidden" in items.raw[item]["flags"]:
+                continue
+        # Keep track of how many signals we've gone through
+        count += 1
+        # Write the stack size signal
+        stack_size = items.raw[item]["stack_size"]
+        combinator.set_signal(index, item, stack_size)
+        index += 1
+        # Once we exceed the current combinator, place it and reset
+        if index == 20:
+            # Add the combinator to the blueprint
+            combinator.id = "{}_{}".format(x, y)
+            blueprint.entities.append(combinator)
+            # Reset the combinator
+            i += 1
+            y = i % 5
+            x = int(i / 5)
+            combinator.set_signals(None)  # Clear signals
+            combinator.tile_position = (x, y)
+            index = 0
+
+    # Add the last combinator if partially full
+    combinator.id = "{}_{}".format(x, y)
+    blueprint.entities.append(combinator)
+
+    # Add connections to each neighbour
+    for cx in range(x):
+        for cy in range(5):
+            here = "{}_{}".format(cx, cy)
+            right = "{}_{}".format(cx + 1, cy)
+            below = "{}_{}".format(cx, cy + 1)
+            try:
+                blueprint.add_circuit_connection("red", here, right)
+            except KeyError:
+                pass
+            try:
+                blueprint.add_circuit_connection("red", here, below)
+            except KeyError:
+                pass
+
+    print("Number of items:", count)
+    print(blueprint.to_string())
+
+
+if __name__ == "__main__":
+    main()
+```
+
+This script is concise and easy to follow, but the really amazing thing about Draftsman is that this script is completely dynamic and can adjust to whatever version of Factorio that you're playing. New items, removed items, or changed stack sizes, regardless of the author of the change, are resolved automatically with the exact same code. For illustration, pictured top-left is what's generated by running the script for a vanilla game, top-right a game with the medium-sized overhaul modpack [Space Exploration](https://mods.factorio.com/mod/space-exploration), and at the bottom a [Bobs](https://mods.factorio.com/user/Bobingabout) + [Pyanodon](https://mods.factorio.com/user/pyanodon) mega-modpack:
+
+![Results](./img/item_stack_size_output.png)
+
+## Image to Blueprint Converter
+
+[This was something I made on a whim.](https://github.com/redruin1/factorio-draftsman/blob/main/examples/image_converter.py) It uses Pillow to load an image and converts it to a blueprint intended to be visible from the map view, all in less than 150 lines of code:
+
+![Portrait of Mike](./img/mike_stoklasa.png "A drunk bum with comically large flask, colorized.")
```

Comment:
Personally I find the description funny, but I think we need to make it a bit more family friendly :)

Reply:
Resolved to `A peculiar man with a comically large flask, colorized.` Let me know if the alcohol is a problem as well.

Reply:
Nah, I think that's perfect. Who knows, could be a coffee flash :P Thanks